optimizely/event/user_event.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2019 Optimizely
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+import uuid
+
+from optimizely import version
+
+SDK_TYPE = 'python-sdk'
+
+
+class UserEvent(object):
+  """ Class respresenting User Event. """
+
+  def __init__(self, event_context):
+    self.event_context = event_context
+    self.uuid = self._get_uuid()
+    self.timestamp = self._get_time()
+
+  def _get_time(self):
+    return int(round(time.time() * 1000))
+
+  def _get_uuid(self):
+    return str(uuid.uuid4())
+
+
+class ImpressionEvent(UserEvent):
+  """ Class representing Impression Event. """
+
+  def __init__(self, event_context, user_id, experiment, visitor_attributes, variation, bot_filtering=None):
+    super(ImpressionEvent, self).__init__(event_context)
+    self.event_context = event_context
+    self.user_id = user_id
+    self.experiment = experiment
+    self.visitor_attributes = visitor_attributes
```

Comment:
Just `self.attributes = attributes`

Reply:
Ali this is visitor attributes, not attributes. Before initializing any impression or conversion object, we filter attributes and convert into `visitor_attributes` otherwise, we'll have dependency while converting into eventbatch object if we use `attributes`